### PR TITLE
Course home: Add icon to switch visibility for all tools in course #3301

### PR DESCRIPTION
**Pull request by carlangas159**
On a course homepage, you cannot change the visibility of all tools at once. Add an open "eye" icon next to the 2 icons of the introduction section, that enable setting all tools to "invisible". Change the icon to closed eye to revert visibility to "visible".

Only show the open eye ("make visible") when all tools are invisible.
Show the closed eye ("make invisible") in any other case

### CODE REVIEW

File: main/inc/ajax/course_home.ajax.php [hunk 1]
```diff
@@ -135,6 +135,139 @@
...
+                            $toolImage,
+                            null,
+                            null,
+                            null,
+                            null,
+                            true
+                        );
+                        $inactiveImage = str_replace('.gif', '_na.gif', $toolImage);
+                    } else {
+                        // Display::return_icon() also checks in the app/Resources/public/css/themes/{theme}/icons folder
```
Comment (jmontoyaa) [inline main/inc/ajax/course_home.ajax.php:208]: Codacy found an issue: [Line exceeds maximum limit of 100 characters; contains 113 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830&bid=18701969)

File: main/course_home/activity.php [hunk 2]
```diff
@@ -52,6 +52,12 @@
 		<div class="alert alert-success" id="id_confirmation_message" style="display:none"></div>
 	</div>';
     $content .= $pluginExtra;
+    //show/hide all tools
+    $content .= '<div class="btn-group pull-right">'.
+        '<a class="hidden invisible-all show-hide-all-tools" href="javascript:void(0);"><img  src="'.api_get_path(WEB_IMG_PATH).'invisible.png" alt="'.get_lang('Deactivate', '').'" title="'.get_lang('Deactivate', '').'"></a>'.
+        '<a class="hidden visible-all  show-hide-all-tools" href="javascript:void(0);"><img  src="'.api_get_path(WEB_IMG_PATH).'visible.png" alt="'.get_lang('Activate', '').'" title="'.get_lang('Activate', '').'"></a>'.
```
Comment (jmontoyaa) [inline main/course_home/activity.php:58]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 219 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/course_home/activity.php [hunk 3]
```diff
@@ -52,6 +52,12 @@
 		<div class="alert alert-success" id="id_confirmation_message" style="display:none"></div>
 	</div>';
     $content .= $pluginExtra;
+    //show/hide all tools
+    $content .= '<div class="btn-group pull-right">'.
+        '<a class="hidden invisible-all show-hide-all-tools" href="javascript:void(0);"><img  src="'.api_get_path(WEB_IMG_PATH).'invisible.png" alt="'.get_lang('Deactivate', '').'" title="'.get_lang('Deactivate', '').'"></a>'.
```
Comment (jmontoyaa) [inline main/course_home/activity.php:57]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 226 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830&bid=18701969)

File: main/inc/lib/course_home.lib.php [hunk 4]
```diff
@@ -909,6 +910,7 @@ public static function show_tools_category($all_tools_list)
                                     ICON_SIZE_SMALL,
                                     false
                                 );
+                                $link['name']='<em id="'.'linktool_'.$tool['iid'].'"class="fa fa-eye-slash text-muted" title="'.get_lang('Activate').'"></em>';
```
Comment (jmontoyaa) [inline main/inc/lib/course_home.lib.php:913]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 159 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/lib/course_home.lib.php [hunk 5]
```diff
@@ -898,6 +898,7 @@ public static function show_tools_category($all_tools_list)
                                     ICON_SIZE_SMALL,
                                     false
                                 );
+                                $link['name']='<em id="'.'linktool_'.$tool['iid'].'"class="fa fa-eye" title="'.get_lang('Deactivate').'"></em>';
```
Comment (jmontoyaa) [inline main/inc/lib/course_home.lib.php:901]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 144 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/introductionSection.inc.php [hunk 6]
```diff
@@ -290,6 +290,13 @@
         api_get_path(WEB_CODE_PATH).'course_info/tools.php?'.api_get_cidreq(),
         ['class' => 'btn btn-default', 'title' => get_lang('CustomizeIcons')]
     );
+
+}
+/* Tool to show /hide all tools on course */
+$toolAllShowHide = '';
+if (api_is_allowed_to_edit() && empty($session_id)) {
+    $toolAllShowHide =  '<button class="btn btn-default hidden visible-all  show-hide-all-tools" title="'.get_lang('Activate', '').'"><em class="fa fa-eye "></em></button>';
+    $toolAllShowHide .=  '<button class="btn btn-default hidden invisible-all show-hide-all-tools" title="'.get_lang('Deactivate', '').'"><em class="fa fa-eye-slash"></em></button>';
```
Comment (jmontoyaa) [inline main/inc/introductionSection.inc.php:299]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 182 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/ajax/course_home.ajax.php [hunk 7]
```diff
@@ -99,6 +99,7 @@
             $requestedVisible = $visibility == 0 ? 1 : 0;
             $requested_view = $visibility == 0 ? 'visible.png' : 'invisible.png';
             $requestedVisible = $visibility == 0 ? 1 : 0;
+            $requested_fa_class = $visibility == 0 ? 'fa fa-eye '.$requested_class : 'fa fa-eye-slash '.$requested_class;
```
Comment (jmontoyaa) [inline main/inc/ajax/course_home.ajax.php:102]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 121 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/ajax/course_home.ajax.php [hunk 8]
```diff
@@ -131,10 +132,146 @@
...
+                            $toolImage,
+                            null,
+                            null,
+                            null,
+                            null,
+                            true
+                        );
+                        $inactiveImage = str_replace('.gif', '_na.gif', $toolImage);
+                    } else {
+                        // Display::return_icon() also checks in the app/Resources/public/css/themes/{theme}/icons folder
```
Comment (jmontoyaa) [inline main/inc/ajax/course_home.ajax.php:210]: Codacy found an issue: [Line exceeds maximum limit of 100 characters; contains 113 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/introductionSection.inc.php [hunk 9]
```diff
@@ -290,6 +290,13 @@
         api_get_path(WEB_CODE_PATH).'course_info/tools.php?'.api_get_cidreq(),
         ['class' => 'btn btn-default', 'title' => get_lang('CustomizeIcons')]
     );
+
+}
+/* Tool to show /hide all tools on course */
+$toolAllShowHide = '';
+if (api_is_allowed_to_edit() && empty($session_id)) {
+    $toolAllShowHide =  '<button class="btn btn-default hidden visible-all  show-hide-all-tools" title="'.get_lang('Activate', '').'"><em class="fa fa-eye "></em></button>';
```
Comment (jmontoyaa) [inline main/inc/introductionSection.inc.php:298]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 173 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

**Comment by jmontoyaa** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830&bid=18701969)

File: main/inc/ajax/course_home.ajax.php (hunk 7 above)
Comment (jmontoyaa) [inline main/inc/ajax/course_home.ajax.php:102]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 121 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/ajax/course_home.ajax.php (hunk 8 above)
Comment (jmontoyaa) [inline main/inc/ajax/course_home.ajax.php:210]: Codacy found an issue: [Line exceeds maximum limit of 100 characters; contains 113 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/lib/course_home.lib.php (hunk 5 above)
Comment (jmontoyaa) [inline main/inc/lib/course_home.lib.php:901]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 144 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/introductionSection.inc.php (hunk 9 above)
Comment (jmontoyaa) [inline main/inc/introductionSection.inc.php:298]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 173 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/lib/course_home.lib.php (hunk 4 above)
Comment (jmontoyaa) [inline main/inc/lib/course_home.lib.php:913]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 159 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

File: main/inc/introductionSection.inc.php (hunk 6 above)
Comment (jmontoyaa) [inline main/inc/introductionSection.inc.php:299]: Codacy found an issue: [Line exceeds maximum limit of 120 characters; contains 182 characters](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=5714830)

**Comment by ywarnier** [comment]
When I hide all tools with the general icon, then I make one single tool visible, the little circle keeps turning and never ends

running-circle

**Comment by ywarnier** [comment]
Merged but there is still this issue to be solved.

**Pull request merged**